File: Python/close_tab_JavascriptExecuter.py
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from selenium.webdriver.chrome.service import Service as ChromeService
from webdriver_manager.chrome import ChromeDriverManager
import time

# Create Chrome Service using the WebDriver Manager
ChromeService = ChromeService(ChromeDriverManager().install())

# Create the Chrome WebDriver
chromeDriver = webdriver.Chrome(service=ChromeService)

# Open "https://ecommerce-playground.lambdatest.io/" web page on the browser
chromeDriver.get("https://ecommerce-playground.lambdatest.io/")

# Print the title of tab in focus
print(chromeDriver.title)

# Open a new tab using the JavascriptExecutor
chromeDriver.execute_script(
    "popup_window = window.open('https://www.google.com')")


# Add time Delay for clear visibility and observability
time.sleep(2)

# Selenium Close tab Python using javascrit executer
chromeDriver.execute_script("popup_window.close()")

# Try to get alert from the web app
alert = driver.switch_to.alert

File: JavaSeleniumCloseTabExample/src/closetab/CloseTabJavascriptExecuter.java
package closetab;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;

// How to close Tab in Selenium Java Example
public class CloseTabJavascriptExecuter {
	public static void main(String[] args) {

		// No need for setProperty as ChromeDriver() will auto install 
		// required browser driver for Chrome using Selenium Manager
		WebDriver driver = new ChromeDriver();
				
		// Open the "https://ecommerce-playground.lambdatest.io/" web page on the browser
		driver.get("https://ecommerce-playground.lambdatest.io/");
			
		// Print the tile of the current tab
		System.out.println(driver.getTitle());
		
		// Creating a JavascriptExecutor reference
        JavascriptExecutor js = (JavascriptExecutor) driver;
		
		// Open a new tab using the JavascriptExecutor
		js.executeScript("popup_window = window.open('https://www.google.com')");
		
		// Print the tile of new tab
		System.out.println(driver.getTitle());
				
		// Add time Delay for page to load 
		try {
			Thread.sleep(2000);
		} catch (InterruptedException e) {
			// Catch block to handle exception
			e.printStackTrace();
		}

        // Selenium close tab using JavascriptExecutor
        js.executeScript("popup_window.close()");
	}
}
